instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/TemporaryMetricsView.java

```diff
@@ -52,10 +66,29 @@
     return view;
   }
 
-  static Attributes applyDurationView(Attributes startAttributes, Attributes endAttributes) {
+  static Attributes applyClientDurationView(Attributes startAttributes, Attributes endAttributes) {
     AttributesBuilder filtered = Attributes.builder();
-    applyView(filtered, startAttributes, durationView);
-    applyView(filtered, endAttributes, durationView);
+    // TODO - Remove query parameters from URL.
+    applyView(filtered, startAttributes, durationClientView);
+    applyView(filtered, endAttributes, durationClientView);
+    return filtered.build();
+  }
+
+  private static <T> boolean containsAttribute(
+      AttributeKey<T> key, Attributes startAttributes, Attributes endAttributes) {
+    return startAttributes.get(key) != null || endAttributes.get(key) != null;
+  }
+
+  static Attributes applyServerDurationView(Attributes startAttributes, Attributes endAttributes) {
+    Set<AttributeKey> fullSet = durationServerView;
+    // Use http.scheme, http.host and http.target when http.route is not available.
+    if (!containsAttribute(SemanticAttributes.HTTP_ROUTE, startAttributes, endAttributes)) {
+      fullSet = new HashSet<>(durationServerView);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    }
```

Comment:
Most of our instrumentations don't set `http.route`. WDYT about creating a static final `durationServerFallbackView` instead of building a new set every time?

Reply:
done!